apps/enc.c

```diff
@@ -601,22 +601,22 @@ static void show_ciphers(const OBJ_NAME *name, void *arg)
         BIO_printf(dec->bio, " ");
 }
 
-static int set_hex(char *in, unsigned char *out, int size)
+static int set_hex(const char *in, unsigned char *out, int size)
 {
     int i, n;
     unsigned char j;
 
+    i = size * 2;
     n = strlen(in);
-    if (n > (size * 2)) {
+    if (n > i) {
         BIO_printf(bio_err, "hex string is too long\n");
-        return 0;
-    }
+        n = i; /* ignore exceeding part */
```

Comment:
I'm not convinced that continuing is a great idea.  At the least, the error message should say thing along the lines of  _hex string is too long, ignoring excess_ so it is completely clear what is happening.

Reply:
You may be right, but please consider the current situation : an error is raised only when iv is longer than the buffer size, not the **expected** iv length (see change line 464).

Reply:
I agree with you about the error message. Thanks.

Reply:
What it does now isn't generally a good indication as to what it should do.  Unfortunately.